File: examples/chat-basic/strava_api.js

require('dotenv').config() 

const auth_link="https://www.strava.com/oauth/token"

function displayActivities(data){
    const column = Object.keys(data[0]) 
    const head = document.querySelector('thead')
    let tags = "<tr>"
    for (i = 0; i<column.length; i++){
        tags += "<th>"+column[i]+"</th>"
    }
    tags += "</tr>"
    head.innerHTML=tags

    const body = document.querySelector('tbody')
    let rows = ""
    data.map(d => {
        rows+= "<tr><td>" + d.start_date + "</td><td>" + d.name + "</td><td>" + d.sport_type + "</td><td>" + Math.round(d.distance)/1000 + "KM</td</tr>"
    })
    head.innerHTML=rows
}

function getActivities(res){
    
    const activities_link = "https://www.strava.com/api/v3/athlete/activities?access_token="+res.access_token
    fetch(activities_link)
        .then((res) => res.json())
            .then(res => displayActivities(res))
    
}

function reAuthorize(){
    fetch(auth_link,
        {
            method:'post',
            headers: {
                'Accept': 'application/json, text/plain, */*',
                'Content-Type': 'application/json'
            },

            body: JSON.stringify(
                {
                    client_id: process.env.CLIENT_ID,
                    client_secret: process.env.CLIENT_SECRET,
                    refresh_token: process.env.REFRESH_TOKEN,
                    grant_type: 'refresh_token'

                }
            )
        }).then(res => res.json())
            .then(res => getActivities(res))
}



//getActivities()
reAuthorize()

File: examples/chat-basic/strava_api.py
import requests
import os
from datetime import datetime, time, date, timezone, timedelta
from dotenv import load_dotenv
load_dotenv()

auth_url="https://www.strava.com/oauth/token"
activities_url = "https://www.strava.com/api/v3/athlete/activities?access_token="

payload={
        'client_id': os.getenv('CLIENT_ID'),
        'client_secret': os.getenv('CLIENT_SECRET'),
        'refresh_token': os.getenv('REFRESH_TOKEN'),
        'grant_type': 'refresh_token',
        'f': 'json'
}

def getStravaData(days):
    res = requests.post(auth_url, data=payload, verify=False)
    access_token = res.json()['access_token']

    header = {'Authorization': 'Bearer ' + access_token}
    param = {'per_page': 200, 'page': 1}
    my_dataset = requests.get(activities_url, headers=header, params=param).json()
    activity_list = []
    for d in my_dataset:
        date = datetime.fromisoformat(
            d['start_date'][:-1]
        )
        date_past = datetime.now() - timedelta(days=days)
        if (date > date_past):
            summary = {
                "Activity":str(d['sport_type']), 
                "Date":str(d['start_date']), #str(date.strftime("%m/%d, %H:%M"))
                "Description":str(d['name']), 
                "Distance (KM)":str(round(d['distance']/1000,2))
            }
            activity_list.append(summary)
    return list(reversed(activity_list))

if __name__ == '__main__':
    getStravaData()
